internal/provider/kubernetes/controller.go

```diff
@@ -198,7 +198,7 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, _ reconcile.Reques
 
 		// Process the parametersRef of the accepted GatewayClass.
 		// This should run before processGateways and processBackendRefs
-		if managedGC.Spec.ParametersRef != nil && managedGC.DeletionTimestamp == nil {
+		if managedGC.Spec.ParametersRef != nil {
```

Comment:
curious why are removing the `DeletionTimestamp` check ?

Reply:
Leftover, I was doing some tests and wanted to avoid the issue of disabling the mergeGateways feature when GatewayClass is being deleted.
Let's restore it and solve it in the next iteration.